packages/pds/src/sequencer/events.ts

```diff
@@ -0,0 +1,144 @@
+import Database from '../db'
+import { z } from 'zod'
+import { cborEncode, schema } from '@atproto/common'
+import {
+  BlockMap,
+  blocksToCar,
+  CidSet,
+  CommitData,
+  WriteOpAction,
+} from '@atproto/repo'
+import { PreparedWrite } from '../repo'
+import { CID } from 'multiformats/cid'
+
+export const sequenceCommit = async (
+  dbTxn: Database,
+  did: string,
+  commitData: CommitData,
+  writes: PreparedWrite[],
+) => {
+  let tooBig: boolean
+  const ops: CommitEvtOp[] = []
+  const blobs = new CidSet()
+  let carSlice: Uint8Array
+
+  // max 200 ops or 1MB of data
+  if (writes.length > 200 || commitData.blocks.byteSize > 1024000) {
+    tooBig = true
```

Comment:
Just my opinion but I think it might be good for compatibility to place these constraints right in the lexicon for the subscription endpoint.  Everyone agreeing on what is "too big" of an event will be useful for consumers.  E.g. if consumers are going to keep a buffer of a fixed size for placing the car, everyone will agree on how big that buffer must be.

Reply:
yup that's a good point 👍